src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -245,17 +245,37 @@ public bool WaitForCompletion(int timeout)
         }
 
         /// <summary>
+        /// Signal any test run that is in process to stop. Return without error if no test is running.
+        /// </summary>
+        /// <param name="force">If true, kill any tests that are currently running</param>
+        public void StopRun(bool force)
+        {
+            if (IsTestRunning)
+            {
+                Context.ExecutionStatus = force
+                    ? TestExecutionStatus.AbortRequested
+                    : TestExecutionStatus.StopRequested;
+
+                Context.Dispatcher.CancelRun(force);
+            }
+        }
+
+        #endregion
+
+        #region Helper Methods
+
+        /// <summary>
         /// Initiate the test run.
         /// </summary>
-        public void StartRun(ITestListener listener)
+        private void StartRun(ITestListener listener)
```

Comment:
Actually, they are not both public. I'm making StartRun, which is a helper method, private. StopRun was always public and still is. The diff is making it look like StartRun is new, but if you look further down, you'll see its location merely changed.